services/brig/src/Brig/API/User.hs

```diff
@@ -1437,6 +1438,30 @@ lookupProfiles self others =
       (lookupProfilesFromDomain self)
       (bucketQualified others)
 
+-- | Similar to lookupProfiles except it returns all results and all errors
+-- allowing for partial success.
+lookupProfilesV3 ::
+  ( Member GalleyProvider r,
+    Member (Concurrency 'Unsafe) r
+  ) =>
+  -- | User 'self' on whose behalf the profiles are requested.
+  Local UserId ->
+  -- | The users ('others') for which to obtain the profiles.
+  [Qualified UserId] ->
+  AppT r ([(Qualified UserId, FederationError)], [UserProfile])
+lookupProfilesV3 self others = do
+  t <-
+    traverseConcurrently
+      (lookupProfilesFromDomain self)
+      (bucketQualified others)
+  let (l, r) = partitionEithers t
+  pure (l >>= flattenUsers, join r)
+  where
+    flattenUsers :: (Qualified [UserId], FederationError) -> [(Qualified UserId, FederationError)]
+    flattenUsers (l, e) =
+      let mkUser u = Qualified u (qDomain l)
+       in (,e) . mkUser <$> qUnqualified l
```

Comment:
While I suppose this is not wrong to unqualify and then to qualify again by hand, how about this?
```haskell
flattenUsers (l,e) = (,e) <$> sequence l
```

Reply:
I've swapped over, using sequenceA